lisa/features/availability.py

```diff
@@ -0,0 +1,172 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+from dataclasses import dataclass, field
+from enum import Enum
+from functools import partial
+from typing import Any, Dict, List, Optional, Type, Union
+
+from dataclasses_json import dataclass_json
+
+from lisa import schema, search_space
+from lisa.feature import Feature
+from lisa.util import LisaException, constants, field_metadata
+
+FEATURE_NAME_AVAILABILITY = "Availability"
+
+
+class AvailabilityType(str, Enum):
+    Default = constants.AVAILABILITY_DEFAULT
+    NoRedundancy = constants.AVAILABILITY_NONE
+    AvailabilitySet = constants.AVAILABILITY_SET
+    AvailabilityZone = constants.AVAILABILITY_ZONE
+
+
+availability_type_priority: List[AvailabilityType] = [
+    AvailabilityType.NoRedundancy,
+    AvailabilityType.AvailabilitySet,
+    AvailabilityType.AvailabilityZone,
+]
+
+
+@dataclass_json()
+@dataclass
+class AvailabilityArmParameter:
```

Comment:
It's Azure specified, so it should be in Azure features file.

Reply:
Isn't the whole concept of availability zones and availability sets specific to Azure? I imagine this feature won't be enabled in any other platform, so what is the purpose of separating logic into different files?

Reply:
Availability may not be a general name, but there is a common concept to run service with high availability. For example, when servers are upgrading for a service, the servers can be upgraded by availability sets. It also works for faults. If the servers are in two sets, the service is continuing. The zones may be redundant for electric incidences. Check details here, https://techcommunity.microsoft.com/t5/itops-talk-blog/understanding-availability-sets-and-availability-zones/ba-p/1992518

Reply:
We can also treat it as an Azure feature, and put it in azure/features like AzureExtensions.

Reply:
I moved AvailabilityArmParameter to common because I couldn't find a way to import it in AzureArmParameter without a circular dependency. Common is also where it is primarily used, so I think this is an acceptable location.

Reply:
It's ok.